pandas/tests/groupby/test_libgroupby.py

```diff
@@ -282,3 +282,20 @@ def test_cython_group_mean_not_datetimelike_but_has_NaT_values():
     tm.assert_numpy_array_equal(
         actual[:, 0], np.array(np.divide(np.add(data[0], data[1]), 2), dtype="float64")
     )
+
+
+def test_cython_group_mean_Inf_at_begining_and_end():
+    # GH 50367
+    actual = np.array([[np.nan, np.nan], [np.nan, np.nan]], dtype="float64")
+    counts = np.array([0, 0], dtype="int64")
+    data = np.array(
+        [[np.inf, 1.0], [1.0, 2.0], [2.0, 3.0], [3.0, 4.0], [4.0, 5.0], [5, np.inf]],
+        dtype="float64",
+    )
+    labels = np.array([0, 1, 0, 1, 0, 1], dtype="int64")
+
+    group_mean(actual, counts, data, labels, is_datetimelike=False)
+
+    tm.assert_numpy_array_equal(
+        actual, np.array([[np.inf, 3], [3, np.inf]], dtype="float64")
```

Comment:
nitpick: please define the expected on a previous line

Reply:
Done